src/index.js

```diff
@@ -15,15 +29,16 @@ export default function register(Component, tagName, propNames, options) {
 	PreactElement.prototype.attributeChangedCallback = attributeChangedCallback;
 	PreactElement.prototype.disconnectedCallback = disconnectedCallback;
 
-	// b \ a
-	// take all Component properties and remove existing PreactElement properties
-	const a = new Set(Object.getOwnPropertyNames(PreactElement.prototype));
-	const b = new Set(Object.getOwnPropertyNames(Component.prototype));
-	const diff = Array.from(new Set([...b].filter((x) => !a.has(x))));
+	const methods = Object.getOwnPropertyNames(Component.prototype)
+		.filter((propertyName) => !lifecycleMethods.includes(propertyName))
+		.filter((propertyName) => !apiMethods.includes(propertyName))
+		.filter(
+			(propertyName) => typeof Component.prototype[propertyName] === 'function'
+		);
```

Comment:
Should we try to combine these into one filter statement?

Reply:
Why? I don't see `register` being a hot codepath.

Reply:
Still saves a few bytes, up to you though :)

Reply:
Normally, I prefer to leave minification and compression to the tooling.